langchain/src/tools/connery.ts

```diff
@@ -0,0 +1,353 @@
+      "Do not come up with the values for the input parameters yourself. " +
+      "If you do not have enough information to fill in the input parameter, ask the user to provide it. " +
+      "See examples below on how to construct the prompt based on the provided tool information. " +
+      "\n\n" +
+      "# Instructions about tool output:\n" +
+      "The output of this tool is a JSON string. " +
+      "Retrieve the output parameters from the JSON string and use them in the next tool. " +
+      "Do not return the JSON string as the output of the tool. " +
+      "\n\n" +
+      "# Example:\n" +
+      "Tool information:\n" +
+      "- Title: Send email\n" +
+      "- Description: Send an email to a recipient.\n" +
+      `- Input parameters schema in JSON fromat: ${example1InputParametersSchema}\n` +
+      "The tool input prompt:\n" +
+      "recipient: test@example.com, subject: 'Test email', body: 'This is a test email sent from Langchain Connery tool.'\n" +
+      "\n\n" +
+      "# The tool information\n" +
+      `- Title: ${title}\n` +
+      `- Description: ${description}\n` +
+      `- Input parameters schema in JSON fromat: ${inputParameters}\n`;
+
+    return descriptionTemplate;
+  }
+
+  /**
+   * Converts the provided object to a JSON string and escapes '{' and '}' characters.
+   * @param obj The object to convert to a JSON string.
+   * @returns A string containing the JSON representation of the provided object with '{' and '}' characters escaped.
+   */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  protected prepareJsonForTemplate(obj: any): string {
+    // Convert the object to a JSON string
+    const jsonString = JSON.stringify(obj);
+
+    // Replace '{' with '{{' and '}' with '}}'
+    const escapedJSON = jsonString.replace(/{/g, "{{").replace(/}/g, "}}");
+
+    return escapedJSON;
+  }
+}
+
+/**
+ * A service for working with Connery actions.
+ *
+ * Connery is an open-source plugin infrastructure for AI.
+ * Source code: https://github.com/connery-io/connery-platform
+ */
+export class ConneryService {
+  protected runnerUrl: string;
+
+  protected apiKey: string;
+
+  protected asyncCaller: AsyncCaller;
+
+  /**
+   * Creates a ConneryService instance.
+   * @param params A ConneryServiceParams object.
+   * If not provided, the values are retrieved from the CONNERY_RUNNER_URL
+   * and CONNERY_RUNNER_API_KEY environment variables.
+   * @returns A ConneryService instance.
+   */
+  constructor(params?: ConneryServiceParams) {
+    const runnerUrl =
+      params?.runnerUrl ?? getEnvironmentVariable("CONNERY_RUNNER_URL");
+    const apiKey =
+      params?.apiKey ?? getEnvironmentVariable("CONNERY_RUNNER_API_KEY");
+
+    if (!runnerUrl || !apiKey) {
+      throw new Error(
+        "CONNERY_RUNNER_URL and CONNERY_RUNNER_API_KEY environment variables must be set."
+      );
+    }
+
+    this.runnerUrl = runnerUrl;
```

Comment:
Is there a default value that makes sense for this?

Reply:
There is no default value. The idea of Connery is that everyone has their own isolated runner with a unique URL where they keep all their actions and connections.